.pre-commit-config.yaml

```diff
@@ -0,0 +1,18 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.0.1
+    hooks:
+      - id: check-docstring-first
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+  - repo: https://github.com/myint/autoflake
+    rev: v1.4
+    hooks:
+      - id: autoflake
+        args: ["--in-place", "--remove-all-unused-imports", "--ignore-init-module-imports", "--remove-unused-variables"]
+  - repo: https://github.com/pycqa/flake8
```

Comment:
How does this hook mechanism work? Does this automatically install stuff from the repos into my current environment? I'd rather have the dependencies set up once and other code such as pre-commit only use what's there.

Reply:
It creates it's own virtual env that gets stored somewhere (not sure where). 

![image](https://user-images.githubusercontent.com/10111092/141196583-995cdc49-d3ec-415d-a5e4-4e31074ead9f.png)

(The warning there is fixed now)

Reply:
Ok.

Reply:
Installs into a local cache for me: `~/.cache/pre-commit`